### PR TITLE
Coerce s n

### PR DESCRIPTION
**Pull request by pmurias**
Make coercing a str to a num behave more like the .Num method.
Make it no longer os dependent.

### CODE REVIEW

File: src/strings/parse_num.h [hunk 1]
```diff
@@ -0,0 +1 @@
+MVMnum64 MVM_coerce_s_n_new(MVMThreadContext *tc, MVMString *s);
```
Comment (zhuomingliang) [inline src/strings/parse_num.h]: `MVM_coerce_s_n` me thinks?

**Comment by pmurias** [comment]
Fixed.

**Comment by pmurias** [comment]
NQP tests where doing some incorrect conversions that where uncovered by this PR.
I fixed those tests.

**Comment by jnthn** [comment]
Finally got around to reviewing this (sorry it took a while), and it looks good overall to me. Release is very soon, so suggest merging this just after that.

**Pull request merged**